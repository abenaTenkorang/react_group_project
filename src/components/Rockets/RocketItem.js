import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import {
  addReservation,
  cancelReservation,
} from '../../store/rocket/rocketSlice';
import './RocketItem.css';

export default function Rocket({ rocket }) {
  const dispatch = useDispatch();

  return (
    <div className="rocket_container">
      <img
        className="img"
        src={rocket.flickr_images[0]}
        alt={rocket.rocket_name}
      />
      <div className="rocket-details">
        <h3 className="name">{rocket.rocket_name}</h3>
        <p className="description">
          {rocket.reserved && <span className="reserve-badge">reserved</span>}
          {rocket.description}
        </p>
        {rocket.reserved && (
          <button
            type="button"
            className="cancel-reserve-btn"
            onClick={() => dispatch(cancelReservation(rocket.id))}
          >
            Cancel Reservation
          </button>
        )}
        {!rocket.reserved && (
          <button
            type="button"
            className="reserve-btn"
            onClick={() => dispatch(addReservation(rocket.id))}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
}

Rocket.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string,
    rocket_name: PropTypes.string,
    reserved: PropTypes.bool,
    description: PropTypes.string,
    flickr_images: PropTypes.arrayOf(PropTypes.string),
  }),
};

Rocket.defaultProps = {
  rocket: {
    id: '',
    rocket_name: '',
    reserved: false,
    description: '',
    flickr_images: [],
  },
};
