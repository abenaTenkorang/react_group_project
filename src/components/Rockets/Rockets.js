import './Rockets.module.css';

function Rockets() {
  return (
    <div>
      <h1>Rockets</h1>
    </div>
  );
}

export default Rockets;
