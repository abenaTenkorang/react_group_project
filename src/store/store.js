import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rocket/rocketSlice';
import missionsReducer from './missions/missions';

export default configureStore({
  reducer: {
    rockets: rocketReducer,
    missions: missionsReducer,
  },
});
