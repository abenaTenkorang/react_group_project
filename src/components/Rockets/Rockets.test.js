import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../store/store';
import Rockets from './Rockets';

describe('Rockets', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Rockets />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
