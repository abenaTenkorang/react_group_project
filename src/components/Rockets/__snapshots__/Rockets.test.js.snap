// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rockets renders correctly 1`] = `
<div>
  <h2
    className="loading"
  >
    Loading...
  </h2>
  <ul
    className="list"
  />
</div>
`;
